pkg/core/intl/oidc.go

```diff
@@ -18,66 +15,46 @@ func LocalizeJSONObject(preferredLanguageTags []string, jsonObject map[string]in
 			continue
 		}
 		if k == key {
-			m[""] = stringValue
+			m[fallbackLanguageTag] = stringValue
 		} else if strings.HasPrefix(k, prefix) {
 			tag := strings.TrimPrefix(k, prefix)
 			m[tag] = stringValue
 		}
 	}
-	_, value := Localize(preferredLanguageTags, m)
+
+	var supportedLanguageTags []string
+	for tag := range m {
+		supportedLanguageTags = append(supportedLanguageTags, tag)
+	}
+	supportedLanguageTags = SortSupported(supportedLanguageTags, fallbackLanguageTag)
+
+	idx, _ := Match(preferredLanguageTags, supportedLanguageTags)
+	tag := supportedLanguageTags[idx]
+	value := m[tag]
 	return value
 }
 
 // LocalizeStringMap returns the localized value of key in stringMap according to preferredLanguageTags.
-func LocalizeStringMap(preferredLanguageTags []string, stringMap map[string]string, key string) string {
+func LocalizeStringMap(preferredLanguageTags []string, fallbackLanguageTag string, stringMap map[string]string, key string) string {
 	prefix := fmt.Sprintf("%s#", key)
 	m := map[string]string{}
 	for k, stringValue := range stringMap {
 		if k == key {
-			m[""] = stringValue
+			m[fallbackLanguageTag] = stringValue
```

Comment:
I found a bug that when I use the default config, this function will return empty string.

When `fallbackLanguageTag` is empty, the map `m` here store default value with empty key. `SortSupported` will populate default `en` to `fallbackLanguageTag` and the matched tag will be `en`.

Reply:
I will add a if statement to set fallbackLanguageTag to DefaultLanguage if it is empty.